crates/sequencer/src/sequencer.rs

```diff
@@ -238,7 +234,23 @@ where
             da_block.header(),
             &mut signed_batch,
         ) {
-            (Ok(()), batch_workspace) => {
+            (Ok(()), mut batch_workspace) => {
+                let system_tx_gas_usage = self
+                    .db_provider
+                    .evm
+                    .get_pending_txs_cumulative_gas_used(&mut batch_workspace);
+
+                let rlp_txs = self.get_best_transactions(l2_block_mode, system_tx_gas_usage);
```

Comment:
instead of passing down l2_block_mode down to get_best_transactions, just set rlp_txs to vec![]